### PR TITLE
Docs: Add CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by dwaynebailey**
This lifts heavily from https://github.com/translate/pootle/wiki/Styleguide-%28proposal%29

I've written this more to provide some guideance to users who will see it in the Github link when they create a new Github issue.  Based in some part on a few unhelpful issues being reported without basic info thus creating the need to ask questions that could have been addressed by the reporter.

Please review this for readability.  The idea is to keep is clean and clear.

I realise it duplicates other stuff, but I'd prefer to have something working now then mash all those fixes together.

If there is something desperate you feel should be here, please flag it, but likley I'll ask that you write that so that it doesn't block this landing.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Translations
+------------
+
+Please translate `online <http://pootle.locamotion.org/projects/pootle/>`_
+
+Need help?
+----------
+
+You can always ask for help in the `#pootle-dev
```
Comment (julen) [inline CONTRIBUTING.rst]: Sounds like routing people to a support channel, which `#pootle-dev` is not.

**Comment by julen** [comment]
The main problem I see here is that it's talking about issues, commits and PRs even before telling people to do their homework: double-check the docs before reporting issues; unless trivial fixes, do not write any code without consulting about it first —even more, and this applies to anyone: think if any code is needed at all ([The Best Code is No Code At All](http://blog.codinghorror.com/the-best-code-is-no-code-at-all/)). This might sound common sense, but practice shows that people not always apply it.

**Comment by unho** [comment]
Just wondering if that doc should just point to http://docs.translatehouse.org/projects/pootle/en/latest/developers/hacking.html

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Commits and pull requests
+-------------------------
+
+Keep the commit log as healthy as the code. It is one of the first places new
+contributors will look at the project.
+
+#. No more than one change per commit. There should be no changes in a commit
+   which are unrelated to its message.
+#. Every commit should pass all tests on its own.
```
Comment (unho) [inline CONTRIBUTING.rst]: Ideally yes, but I am ok with just asking the PR to pass all tests.

**Comment by unho** [comment]
Replaced by PR #4228. Requested changes on this PR have been applied there. So closing this one.

**Pull request closed without merging**